mir_eval/display.py

```diff
@@ -0,0 +1,672 @@
+        seg_map[lab].update(kwargs)
+        seg_map[lab]['label'] = lab
+
+    for ival, lab in zip(intervals, labels):
+        rect_kwargs = seg_map[lab]
+        rect = Rectangle((ival[0], base), ival[1] - ival[0], height,
+                         **rect_kwargs)
+        ax.add_patch(rect)
+        rect_kwargs.pop('label', None)
+
+        if text:
+            ann = ax.annotate(lab,
+                              xy=(ival[0], height), xycoords='data',
+                              xytext=(8, -10), textcoords='offset points',
+                              **text_kw)
+            ann.set_clip_path(rect)
+
+    ax.set_yticks([])
+
+    __expand_limits(ax, [intervals.min(), intervals.max()], which='x')
+
+    return ax
+
+
+def labeled_intervals(intervals, labels, label_set=None,
+                      base=None, height=None, extend_labels=True,
+                      ax=None, tick=True, **kwargs):
+    '''Plot labeled intervals with each label on its own row.
+
+    Parameters
+    ----------
+    intervals : np.ndarray, shape=(n, 2)
+        segment intervals, in the format returned by
+        :func:`mir_eval.io.load_intervals` or
+        :func:`mir_eval.io.load_labeled_intervals`.
+
+    labels : list, shape=(n,)
+        reference segment labels, in the format returned by
+        :func:`mir_eval.io.load_labeled_intervals`.
+
+    label_set : list
+        An (ordered) list of labels to determine the plotting order.
+        If not provided, the labels will be inferred from existing
+        `yticklabels`.
+        If no `yticklabels` exist, then the sorted set of unique values
+        in `labels` is taken as the label set.
+
+    base : np.ndarray, shape=(n,), optional
+        Vertical positions of each label
+
+    height : np.ndarray, shape=(n,), optional
+        Height for each label
+
+    extend_labels : bool
+        If `False`, only values of `labels` that also exist in `label_set`
+        will be shown.
+
+        If `True`, all labels are shown, with those in `labels` but
+        not in `label_set` appended to the top of the plot.
+        A horizontal line is drawn to indicate the separation between
+        values in or out of `label_set`.
+
+    ax : matplotlib.pyplot.axes
+        An axis handle on which to draw the intervals.
+        If none is provided, a new set of axes is created.
+
+    tick : bool
+        If `True`, sets tick positions and labels on the y-axis.
+
+    kwargs
+        Additional keyword arguments to pass to
+        `matplotlib.patches.Rectangle`.
+
+    Returns
+    -------
+    ax : matplotlib.pyplot.axes._subplots.AxesSubplot
+        A handle to the (possibly constructed) plot axes
+    '''
+
+    if ax is None:
+        # Create a new axis
+        ax = plt.gca()
+
+    if label_set is None:
+        # If we have non-empty pre-existing tick labels, use them
+        label_set = [_.get_text() for _ in ax.get_yticklabels()]
+        if not any(label_set):
```

Comment:
Add a comment here, e.g. "When all of the labels are empty strings, treat as an empty label set"